src/metabase/query_processor/preprocess.clj

```diff
@@ -135,10 +136,12 @@
               (when-not (= <> query)
                 (let [middleware-fn-name (if-let [fn-name (:name (meta middleware-fn))]
                                            (if-let [fn-ns (:ns (meta middleware-fn))]
-                                             (format "%s/%s" (ns-name fn-ns) fn-name)
+                                             (symbol (format "%s/%s" (ns-name fn-ns) fn-name))
                                              fn-name)
                                            middleware-fn)]
-                  (list middleware-fn-name '=> <>))))
+                  (list middleware-fn-name '=> <>
+                        ^{:portal.viewer/default :portal.viewer/diff}
```

Comment:
Doesn't seem to work, maybe only metadata on the top-level thing that gets tapped is respected idk. Will look into this more soon